qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -90,14 +82,18 @@ def sync_all(self):
             sync_blocks(self.nodes)
             sync_mempools(self.nodes)
 
-    def join_network(self):
-        """
-        Join the (previously split) network halves together.
-        """
+    def join_network(self, id = 1):
+        # Join the (previously split) adjanced nodes in linear network together: 0-1 >-< 2-3
         assert self.is_network_split
-        stop_nodes(self.nodes)
-        wait_bitcoinds()
-        self.setup_network(False)
+        connect_nodes_bi(self.nodes, id, id + 1)
+        self.is_network_split = False
+
+    def join_network_sync(self, id = 1):
+        # Join the (previously split) adjanced nodes in linear network together: 0-1 >-< 2-3 and syncs both parts
+        assert self.is_network_split
+        connect_nodes_bi(self.nodes, id, id + 1)
+        self.sync_all()
+        self.is_network_split = False
```

Comment:
I agree with @a-petrini. This code, in its current form, is not actually synchronizing the entire network, but only nodes `0-1` and `2-3`. With the following modification, you should synchronize the entire network instead:
```suggestion
        self.is_network_split = False
        self.sync_all()
```

Reply:
Done